Sources/NIOHTTP2/ConnectionStateMachine/ConnectionStreamsState.swift

```diff
@@ -333,11 +336,22 @@ struct ConnectionStreamState {
         switch streamID.mayBeInitiatedBy(.client) {
         case true where streamID > self.lastClientStreamID,
             false where streamID > self.lastServerStreamID:
-            // The stream in question is idle.
-            return .connectionError(
-                underlyingError: NIOHTTP2Errors.noSuchStream(streamID: streamID),
-                type: .protocolError
-            )
+            if isLocallyQuiescing {
```

Comment:
Ok so this is looking a lot better. I wonder now only if we need some amount of DoS protection. Presumably we shouldn't tolerate this happening too often. Can we record some counter or something that lets us produce a rate limit elsewhere?

Reply:
This gets ... interesting. 

One approach is to re-use the rate limiter from `DOSHeuristics` within the state machine and check the rate for this specific scenario. This would mean the state machine would need to understand time and would likely become generic over `NIODeadlineClock`. I don't like that at all!

Another approach would be to widen the scope of the DOS scenario we're considering and apply rate limiting in the channel handler as we currently do. I think the appropriate thing to rate limit here is "inbound stream errors". WDYT?